Auto-generated download page

lucperkins opened the pull request:
@yurishkuro This PR adds a downloads page that's generated from a simple YAML source. You can see a prototype of the page here:

https://deploy-preview-144--jaegertracing.netlify.com/download

This is still a WIP so feel free to make suggestions.

yurishkuro commented on content/download/_index.md:
```diff
@@ -0,0 +1,20 @@
+
+Jaeger can be downloaded in two ways:
+
+* As a [static binary](#binaries)
```
As executable binaries

yurishkuro commented on content/download/_index.md:
```diff
@@ -0,0 +1,20 @@
+Jaeger can be downloaded in two ways:
+
+* As a [static binary](#binaries)
+* As a [Docker image](#docker-images)
```
As Docker images

yurishkuro commented on content/download/_index.md:
```diff
@@ -0,0 +1,20 @@
+title: Download Jaeger
+---
+
+Jaeger can be downloaded in two ways:
```
Jaeger components can be ...

yurishkuro commented on data/download.yaml:
```diff
@@ -0,0 +1,42 @@
+docker:
+- name: jaeger-agent
+  description: The Jaeger agent
+- name: all-in-one
```
I would move it to the last place

yurishkuro commented on data/download.yaml:
```diff
@@ -0,0 +1,42 @@
+- name: jaeger-agent
+  description: The Jaeger agent
+- name: all-in-one
+  description: The "all-in-one" hot rod example used in the [getting started](/docs/latest/getting-started) guide
```
We explain all-in-one in Getting Started as "designed for quick local testing, launches the Jaeger UI, collector, query, and agent, with an in memory storage component."  I don't think it makes sense to refer to the HotROD in the description, as the purpose is just having a single binary to run full Jaeger.

yurishkuro commented on data/download.yaml:
```diff
@@ -0,0 +1,42 @@
+- name: jaeger-collector
+  description: The Jaeger collector
+- name: jaeger-query
+  description: The Jaeger query image
```
should this table be titled "for example"? Because it misses many other Docker images like HotROD, Spark job, DB schema utils, etc. If we do want to list everything (which wouldn't be too bad since our DockerHub doesn't have much documentation), then the question is whether the explanation should be elsewhere. The archives with binaries, for example, do not include many of those additional components.

lucperkins replied:
That's up to you. Perhaps we can merge this and you can follow up by adding the images and descriptions that you'd want to see here?

yurishkuro commented on data/download.yaml:
```diff
@@ -0,0 +1,42 @@
+    windows:
+      name: Windows
+      icon: fa-windows
+  fullReleaseVersions:
```
the buttons look great, but I am reluctant to start listing all versions, including minor patches. That sounds like a lot of maintenance work for little value since the versions can easily be seen from the actual download pages (e.g. GitHub Releases). There's also ambiguity as we are changing which binaries are being included in the archives, e.g. HotROD was not built as executable previously.

lucperkins replied:
Related to the comment above, perhaps we can merge this but make the download page a draft, then you can adjust the versions, images, etc. and then remove the draft status of the page?

yurishkuro commented on data/download.yaml:
```diff
@@ -0,0 +1,42 @@
+  - "1.4.1"
+  - "1.5.0"
+  - "1.6.0"
+  partialReleaseVersions:
```
these ones have almost no value since the archives are just the source code, we only started packaging executables in from 1.4.1

lucperkins replied:
Okay, I can remove the "partial release" versions. I assumed you may not want them but figured I'd err on the side of including first and potentially removing later.

lucperkins commented:
@yurishkuro Okay, I've updated some of the wording and removed the older, source-only versions. Let me know if there are any other versions you'd like to remove.

As for the Docker images, either let me know which ones you'd like me to include *or* of course feel free to change the list on your own in a later PR.

yurishkuro commented:
Thanks, @lucperkins. I merged as is, and booked #147 for additional minor fixes.
